g3/generate/prompts/commit/creator.py

```diff
@@ -5,58 +5,93 @@
+                       "We defined a good commit message as one that explains what was changed, and if usefull, why" \
+                       " the change was made. Don't include the provided code in your response or the words commit " \
+                       "message. Be brief on your response using less than 7-10 words, use imperative and start " \
+                       "with a verb. The goal is to briefly complete a sentence which starts like 'This commit if " \
+                       "applied will:' but dont include this sentence in your response. " \
+                       "Whatever follows should be the commit message that you suggest. " \
+                       f"Include in your reponse the following: {include}. " if include else "" \
+                        f"Incorporate the following Jira ticket: {jira}. " if jira else "" \
+                        f"Use the following tone when creating the commit message: {tone}. " \
+                        "Your response should be in markdown format." \
+            }]
+        return messages
+
+    def create_example_message(self, code, commit_creation_messages):
+        code_stack = self.get_code_stack(code)
+        sample = self.set_sample(code_stack)
+        if not sample:
+            return
+
+        commit_creation_messages.append({
+            "role": "user",
+            "content": f"Please provide a commit message for the provided code. Code: ```{sample.get('code')}```"
+        }, {
+            "role": "assistant",
+            "content": sample.get("message")
+        })
+
+    def get_code_stack(self, code: str) -> str:
+        messages = [{
+            "role": "system",
+            "content": "You are a helpful assistant which reads changes from a git repository and returns the "
+                       "most common technology stack used in the code changes provided. "
+                       "The technology stack is defined as the programming language "
+                       "used in the repository. The code changes provided whill contain the names of the files "
+                       "and changes from the result fo the git show result command. "
+                       "The goal is to return the most common stack used in the changes ignoring changes in SQL. "
+                       "Your response should be one of the following: python, ruby, node, or none."
+                       "Don't include the provided code in your response. Provide just one of the options which were"
+                       " given."
+        },
             {
                 "role": "user",
-                "content": f"Please provide a commit message for the provided code. Code: ```{code}```"
+                "content": f"Please provide the technology stack used in the following code changes: ```{code}```"
             }
         ]
         openai_response = openai.ChatCompletion.create(model="gpt-4", messages=messages, temperature=0)
```

Comment:
Actually it would make more sence we return the prompt `Dict[str, str]` and let another class call GPT